pkg/noderesourcetopology/filter.go

```diff
@@ -77,20 +77,20 @@ func resourcesAvailableInAnyNUMANodes(logKey string, numaNodes NUMANodeList, res
 			continue
 		}
 
-		isNodeResource := false
```

Comment:
We just inline this 3 lines of code in previous commit. Why we're moving it back into a function again?
if that's the case, let's remove the commit with the inline work

Reply:
there are two questions here:
1. why function -> inline -> another function? to illustrate the transformative step and to make each step as obvious as possible, providing a rationale for each step. I expect this PR to ber squashed past review anyway, as it is pretty much the standard in this project
2. should this code be inlined or in a function? I actually don't have strong feelings here. I just need we should try very hard to be consistent between the node suitable and the numa zone suitable check: if one is a function, the other should must likely be as we well, bar glaring obvious reasons. This is for consistency. So, `isNUMANodeSuitable` is a function, then I'm making as final state `isNodeSuitable` a function as well. But indeed, it's debatable.

Reply:
thinking more about it, I can do a bit better and avoid the ping pong here. Thanks for raising.